### PR TITLE
Fix water balance error for grass when use FATES-Hydro

### PR DESCRIPTION
**Pull request by JunyanDing**
### Description:

Changed the FatesPlantHydraulicsMod.F90, when the PFT is grass, pool the storage of leaf and stem

### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the in-code documentation .AND. (the [technical note](https://github.com/NGEET/fates-docs) .OR. the wiki) accordingly.
- [ x] I have read the [**CONTRIBUTING**](https://github.com/NGEET/fates/blob/master/CONTRIBUTING.md) document.
- [ ] FATES PASS/FAIL regression tests were run
- [ ] If answers were expected to change, evaluation was performed and provided

### Test Results:

CTSM (or) E3SM (specify which) test hash-tag:

CTSM (or) E3SM (specify which) baseline hash-tag:

FATES baseline hash-tag:

Test Output:

### CODE REVIEW

File: biogeophys/FatesPlantHydraulicsMod.F90 [hunk 1]
```diff
@@ -934,8 +937,16 @@ subroutine UpdatePlantHydrLenVol(ccohort,site_hydr)
...
-    ccohort_hydr%v_ag(n_hypool_leaf+1:n_hypool_ag) = v_sapwood / n_hypool_stem
 
+    ! Junyan changed the following code to calculate the above ground node volume
+    ! foliage donate half of its water volume to xylem for grass
+    if (prt_params%woody(ft)==1) then
+      ccohort_hydr%v_ag(n_hypool_leaf+1:n_hypool_ag) = v_sapwood / n_hypool_stem  ! original code
+    else
+      v_leaf_donate(1:n_hypool_leaf) = ccohort_hydr%v_ag(1:n_hypool_leaf) * l2sap_vol_donate_frac
+      ccohort_hydr%v_ag(1:n_hypool_leaf) = ccohort_hydr%v_ag(1:n_hypool_leaf) - v_leaf_donate(1:n_hypool_leaf)
+      ccohort_hydr%v_ag(n_hypool_leaf+1:n_hypool_ag) = (v_sapwood + sum(v_leaf_donate(1:n_hypool_leaf))) / n_hypool_stem
```
Comment (rgknox) [inline biogeophys/FatesPlantHydraulicsMod.F90:948]: ok, this makes sense

File: parameter_files/fates_params_default.cdl [hunk 2]
```diff
@@ -1445,7 +1445,7 @@ data:
 
  fates_maintresp_model = 1 ;
 
- fates_mort_disturb_frac = 1 ;
```
Comment (rgknox) [inline parameter_files/fates_params_default.cdl:1448]: @JunyanDing can we revert this change? If we change the default file for a non-hydro parameter, I would like to see that change in a separate pull request.

File: parameter_files/fates_params_default.cdl [hunk 3]
```diff
@@ -769,7 +769,7 @@ data:
  fates_history_sizeclass_bin_edges = 0, 5, 10, 15, 20, 30, 40, 50, 60, 70,
     80, 90, 100 ;
 
- fates_hydr_htftype_node = 1, 1, 1, 1 ;
+ fates_hydr_htftype_node = 2, 2, 2, 2 ;
```
Comment (rgknox) [inline parameter_files/fates_params_default.cdl:772]: @JunyanDing do you recommend that most people use type 2, or is this just for your runs?  If we do switch the default, I'd like perspectives from  @pnlfang @xuchongang and others.

File: biogeochem/EDPhysiologyMod.F90 [hunk 4]
```diff
@@ -910,9 +910,10 @@ subroutine phenology( currentSite, bc_in )
     ! and thus %nchilldays will never go from zero to 1.  The following logic
     ! when coupled with this fact will essentially prevent cold-deciduous
     ! plants from re-emerging in areas without at least some cold days
-
+    
     if( (currentSite%cstatus == phen_cstat_notcold)  .and. &
-         (dayssincecleafoff > 400)) then           ! remove leaves after a whole year
+         (dayssincecleafoff > 40000)) then           ! remove leaves after a whole year, 
+                                                     !Junyan change dayssincecleafoff > 400 to 40000 to disable this function, so the condition will never be met
```
Comment (rgknox) [inline biogeochem/EDPhysiologyMod.F90:916]: @JunyanDing this looks like a way of preventing "never cold" status.  Is this something particular to your simulation needs, or is there something flawed with how we do "never cold"?

File: parameter_files/fates_params_default.cdl (hunk 3 above)
Comment (JunyanDing) [reply parameter_files/fates_params_default.cdl:772]: This is just my preference. We can keep the default to be 1

**Comment by niuhanlin** [comment]
The number of iterations is too many and the exit error is reported in the grass when use FATES-Hydro.

File: biogeochem/EDPhysiologyMod.F90 (hunk 4 above)
Comment (ckoven) [reply biogeochem/EDPhysiologyMod.F90:916]: @JunyanDing and I are chatting right now.  We both agree that these phenology changes should be reverted out of this PR.

File: parameter_files/fates_params_default.cdl (hunk 2 above)
Comment (ckoven) [reply parameter_files/fates_params_default.cdl:1448]: Chatting with @JunyanDing, we both agree to revert this change too.

File: biogeochem/EDPhysiologyMod.F90 [hunk 5]
```diff
@@ -815,8 +815,8 @@ subroutine phenology( currentSite, bc_in )
     endif
     !
     ! accumulate the GDD using daily mean temperatures
-    ! Don't accumulate GDD during the growing season (that wouldn't make sense)
-    if (temp_in_C .gt. 0._r8 .and. currentSite%cstatus == phen_cstat_iscold) then
+    ! Don't accumulate GDD during the growing season (that wouldn't make sense)  Junyan modified to adding nevercold 
+    if (temp_in_C .gt. 0._r8 .and. (currentSite%cstatus == phen_cstat_iscold .or. currentSite%cstatus == phen_cstat_nevercold)) then
```
Comment (glemieux) [inline biogeochem/EDPhysiologyMod.F90:819]: Note: Revert this as well.

Comment (glemieux) [reply biogeochem/EDPhysiologyMod.F90]: Done

File: biogeochem/EDPhysiologyMod.F90 (hunk 4 above)
Comment (glemieux) [reply biogeochem/EDPhysiologyMod.F90]: Done

File: parameter_files/fates_params_default.cdl (hunk 3 above)
Comment (glemieux) [reply parameter_files/fates_params_default.cdl]: Reverted.

File: parameter_files/fates_params_default.cdl (hunk 2 above)
Comment (glemieux) [reply parameter_files/fates_params_default.cdl]: Reverted

**Comment by glemieux** [comment]
Regression testing on cheyenne underway.

**Comment by glemieux** [comment]
Regression testing on cheyenne returns B4B results for all non-hydro tests as expected.  Both hydro tests DIFFs look reasonable given that the initial differences are in rootuptake output variables.  Issue #701 appears to have been corrected here in that the exact restart hydro test is passing now `COMPARE_BASE_REST` now.

Folder location: `/glade/u/home/glemieux/scratch/ctsm-tests/tests_pr910`

One note: for some reason `FATES_LAI` was not included in the previous baseline, `fates-sci.1.64.0_api.25.1.0-ctsm5.1.dev118`, for some of the testmods resulting in `FIELDLIST` diffs.  I'm currently looking into why that is and will regenerate

File: main/FatesHydraulicsMemMod.F90 [hunk 6]
```diff
@@ -29,7 +29,7 @@ module FatesHydraulicsMemMod
    integer, parameter, public                  :: n_hypool_stem  = 1
    integer, parameter, public                  :: n_hypool_troot = 1 ! CANNOT BE CHANGED
    integer, parameter, public                  :: n_hypool_aroot = 1 ! THIS IS "PER-SOIL-LAYER"
-   integer, parameter, public                  :: nshell         = 5
+   integer, parameter, public                  :: nshell         = 1
```
[the remark this answers is not in the document]
Comment (rgknox) [reply main/FatesHydraulicsMemMod.F90:32]: checked in via email with @xuchongang and @bchristo and they both blessed this change as well

**Review by rgknox — APPROVED** [review]

**Comment by glemieux** [comment]
> ... Issue #701 appears to have been corrected here, at least temporarily, in that the exact restart hydro test is passing now `COMPARE_BASE_REST` now.

This looks like its a tempermental issue and not necessarily fixed by this PR.  In re-running the previous baseline, this issue wasn't present suggesting that there is something else going on to cause #701.

**Comment by glemieux** [comment]
> One note: for some reason `FATES_LAI` was not included in the previous baseline, `fates-sci.1.64.0_api.25.1.0-ctsm5.1.dev118`, for some of the testmods resulting in `FIELDLIST` diffs. I'm currently looking into why that is and will regenerate

This baseline has been regenerated and tested against the old version (which I've saved with the suffix `-nolai`).  Everything was B4B except that the regenerated baseline has `FATES_LAI` reinstated.  Still not sure why that happened.
